tinyagentos/routes/settings.py

```
@@ -568,6 +568,13 @@ async def apply_update(request: Request):
     )
     await proc2.communicate()
 
+    # Rebuild the desktop bundle if new source files landed (belt-and-braces on
+    # non-systemd hosts where ExecStartPre doesn't run: Mac .app, Docker, dev).
+    from tinyagentos.desktop_rebuild import rebuild_desktop_bundle_if_stale
+    _rebuilt, _rebuild_msg = await rebuild_desktop_bundle_if_stale(project_dir)
+    if _rebuilt:
+        logger.info("Desktop rebuild: %s", _rebuild_msg)
```

Ruff (0.15.12)

[error] Undefined name `logger`

(F821)